scripts/init-repo.ps1

```diff
@@ -0,0 +1,104 @@
+<#
+.DESCRIPTION
+Initialize zure DevOps GIT repository with branches, policies, rights, build pipelines..see https://github.com/Oriflame/devops/blob/master/docs/init-repo.md
+
+.PARAMETER AzureDevOpsCollection
+Url of Azure DevOps organisation, e.g. https://dev.azure.com/organisationName/
+
+.PARAMETER DevOpsProject
+Name of the DevOps project, e.g. http://dev.azure.com/oriflame/{THIS}
+
+.PARAMETER RepositoryName
+Name of the GIT Repository, e.g. http://dev.azure.com/oriflame/something/_git/{THIS}
+
+.PARAMETER dependenciesRepositoryUrl
+Url of dependend scripts and resources, e.g. url from GitHub
+
+#>
+[cmdletbinding()]
+param
+(
+    [string]
+    $AzureDevOpsCollection = "https://dev.azure.com/oriflame/",
+
+    [string]
+    $DevOpsProject,
+
+    [string]
+    $RepositoryName,
+
+    [string]
+    $DependenciesRepositoryUrl = "https://raw.githubusercontent.com/Oriflame/devops/master/scripts/init-repo/"
+)
+$ErrorActionPreference='Stop';
+$Error.Clear();
+$LASTEXITCODE = 0;
+
+Set-ExecutionPolicy Bypass -Scope Process -Force;
+
+function GetResource ([string]$nameOfScript)
+{
+    $url = "${DependenciesRepositoryUrl}$nameOfScript";
+    Write-Verbose "Downloading resource $url";
+    return (New-Object System.Net.WebClient).DownloadString($url);
+}
+
+#import functions from remote/local repo
+iex (GetResource -nameOfScript branchpermissions.ps1)
+iex (GetResource -nameOfScript branchpolicies.ps1)
+iex (GetResource -nameOfScript cibuild.ps1)
+iex (GetResource -nameOfScript functions.ps1)
+iex (GetResource -nameOfScript gitrepo.ps1)
+iex (GetResource -nameOfScript prerequisities.ps1)
```

Comment:
Invoking an expression with string downloaded froma remote location (that is not under control of a user) smells a little bit like security vulnerability. I understand that it simplifies distributing of all the scripts. However, it forces a user to run it with elevated permissions. As a proove of concept it is OK but I think we should wrap a logix into a PS module and publish it to PS gallery as a final solution. WDYT?

Reply:
Well I wanted to have a one single line of code for an installation, downloadable from a remote repository.  This is why I used the same way as e.g. Chocolatey (Invoke-Expression). It does not force the user to run under elevated permissions. Even Chocolatey can be installed under [non-admin user](https://chocolatey.org/docs/installation#non-administrative-install). Maybe just TF.EXE and CLI is not supported as non-admin install, yet those can be installed manually.

However we can file a suggestion for publishing and using powershell to the PS gallery, this is an idea we should at least think of.